src/ast/node.h

```diff
@@ -29,6 +30,9 @@ struct donsus_node_type {
 
 std::string de_get_from_donsus_node_type(donsus_node_type type);
 
+// Forward declaration of the node structure
```

Comment:
I suspect this is correct, although it might be tricky if the compiler needs more information.

- see: https://stackoverflow.com/questions/10894804/forward-declaration-of-struct